### ISSUE TITLE
java.lang.UnsupportedClassVersionError

### ISSUE THREAD

**Issue by EmilVogt** [valentjn/vscode-ltex#38]
I just found your extension and I'm glad someone is keeping a language tool extension up to date :)

**Describe the bug**
Nothing happens, no sign of any activity of the extension

**Steps to reproduce**

1. Install extension
2. Install English support
3. Restart VS Code
4. Open .tex document
5. Nothing happens

**Expected behavior**
Latex document is spell and grammar checked.

**LTeX configuration**
`"ltex.trace.server": "verbose"`

**Log files**
```
java.lang.UnsupportedClassVersionError: App has been compiled by a more recent version of the Java Runtime (class file version 55.0), this version of the Java Runtime only recognizes class file versions up to 52.0
	at java.lang.ClassLoader.defineClass1(Native Method)
	at java.lang.ClassLoader.defineClass(Unknown Source)
	at java.security.SecureClassLoader.defineClass(Unknown Source)
	at java.net.URLClassLoader.defineClass(Unknown Source)
	at java.net.URLClassLoader.access$100(Unknown Source)
	at java.net.URLClassLoader$1.run(Unknown Source)
	at java.net.URLClassLoader$1.run(Unknown Source)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.net.URLClassLoader.findClass(Unknown Source)
	at java.lang.ClassLoader.loadClass(Unknown Source)
	at sun.misc.Launcher$AppClassLoader.loadClass(Unknown Source)
	at java.lang.ClassLoader.loadClass(Unknown Source)
	at sun.launcher.LauncherHelper.checkAndLoadMain(Unknown Source)
Error: A JNI error has occurred, please check your installation and try again
Exception in thread "main"
```

**Version information**

- Operating system: Windows 10 (1909)
- VS Code: 1.44.2
- LTeX: 4.9.2
- LTeX language extensions: English
- Java: 1.8.0_251

**Comment by EmilVogt** [comment]
Do I have to install the jdk from [https://jdk.java.net/](https://jdk.java.net/)?

I only have the one at [https://www.java.com/en/download/windows-64bit.jsp](https://www.java.com/en/download/windows-64bit.jsp).

Oh, there's also [https://www.oracle.com/Java/technologies/javase-downloads.html](https://www.oracle.com/Java/technologies/javase-downloads.html).

**Comment by valentjn** [comment]
@EmilVogt Thanks for the report. Since 4.9.2, LT<sub>E</sub>X requires Java 11. You have Java 8 (see your second link), so that explains the error.

The thing is that Oracle recently changed how they distribute Java. Previously, mainly Oracle provided a Java distribution (JRE and JDK). That changed to a multi-vendor approach, so now there are lots of different Java distributions. Most of them build on OpenJDK, an open-source reference implementation of Java. OpenJDK also distributes binaries called Oracle OpenJDK (see your first link), but no Windows installers. So it's possible to use that, but a bit tedious. For Windows, I'd recommend [AdoptOpenJDK](https://adoptopenjdk.net/).

Your third link is Oracle's closed-source non-free distribution of the OpenJDK called Oracle JDK. It's free of charge for personal use, but not for commercial use.

The whole situation is pretty confusing, but as long as you install a distribution that supports at least Java 11, you should be good to go. For more information, read [Java is Still Free 2.x](https://medium.com/@javachampions/java-is-still-free-2-0-0-6b9aa8d6d244).

**Comment by valentjn** [comment]
This info should be in the readme. Currently, it links to java.com, where only Java 8 is available. Confirming as a documentation bug.

**Comment by universemaster** [comment]
I think this update is going to break ltex for a lot of people and have them waste a lot of time.  I've just spent 20 minutes uninstalling and reinstalling java from java.com.  I'm pretty literate in these things, but I've still managed to waste a lot of time on this.  Installing jdk 14 from https://jdk.java.net/ now.

**Comment by valentjn** [comment]
@universemaster It seems I underestimated how many people still run Java 8. Oracle will be dropping support by the end of this year, and large projects such as the Eclipse platform will be moving to Java 11 pretty soon. So in the not-so-long term, there's no other option for LT<sub>E</sub>X to move to Java 11 as well.

As a short-term solution, I'll revert the changes made in LT<sub>E</sub>X 4.9.2, so Java 8 will suffice again, while we figure out a transition plan. I'm open to any kind of constructive ideas. (Maybe this is motivation for someone to tackle issue #5.)

**Comment by valentjn** [comment]
Fix released in 4.9.3.
